components/dashboard/src/service/service.tsx

```diff
@@ -21,13 +21,21 @@ import { IDEFrontendDashboardService } from "@gitpod/gitpod-protocol/lib/fronten
 import { RemoteTrackMessage } from "@gitpod/gitpod-protocol/lib/analytics";
 import { helloService } from "./public-api";
 import { getExperimentsClient } from "../experiments/client";
+import { ConnectError, Code } from "@bufbuild/connect";
+import { instrumentWebSocket } from "./metrics";
 
 export const gitpodHostUrl = new GitpodHostUrl(window.location.toString());
 
 function createGitpodService<C extends GitpodClient, S extends GitpodServer>() {
     let host = gitpodHostUrl.asWebsocket().with({ pathname: GitpodServerPath }).withApi();
 
     const connectionProvider = new WebSocketConnectionProvider();
+    const originalCreateWebSocket = connectionProvider["createWebSocket"];
```

Comment:
nit: Can we wrap this into a function `instrumentWebSocketConnection` below?
I feel this file is getting a bit messy lately :sweat_smile: